### ISSUE TITLE
The problem with importing vehicles

### ISSUE THREAD

**Issue by IdleSol** [CleverRaven/Cataclysm-DDA#76687]
### Describe the bug

I will cite the error itself below. But it is not important. The problem is in the order of reading the json file created when exporting transport. If I understand correctly.

The situation is as follows. There is a truck in the game. It is possible to spawn it. There will be no errors. It can be exported to a file. Then put this file in the json folder and get an error preventing loading.

```
 DEBUG    : installing hdframe would make invalid vehicle: Part needs to be adjacent to or on existing structure.

 FUNCTION : int vehicle::install_part(const point&, vehicle_part&&)
 FILE     : src/vehicle.cpp
 LINE     : 1494
 VERSION  : cdda-experimental-2024-09-27-0529 918a370

 DEBUG    : init_vehicles: 'TEST' part 'hdframe'(31) can't be installed to -3,-1

 FUNCTION : void vehicles::finalize_prototypes()
 FILE     : src/veh_type.cpp
 LINE     : 1559
 VERSION  : cdda-experimental-2024-09-27-0529 918a370
```

### Attach save file

[123.json](https://github.com/user-attachments/files/17165081/123.json)

### Steps to reproduce

1. Standard world and character
2. Debug menu - `s`pawning - `v`ehicle - Semi Truck (semi_truck) - undamaged
![1](https://github.com/user-attachments/assets/ff2290e2-22ff-493c-9c92-95e1cb282244)

3. Debug menu - `v`ehicle... - `e`xport vehicle as prototype - 123
4. Quit the game
4. Move the 123.json file from the ./export_dir folder to the ./data/json folder
5. Change the `id` and `name` of the transport (optional)
```diff
...
--  "id":"/TO_BE_REPLACED/",
++  "id":"test_123",
  "type":"vehicle",
--  "name":"/TO_BE_REPLACED/",
++  "name":"test123",
...
```
6. Save file
7. Load a new world/save
8. Get an error message
9. `I`gnore error
10. Debug menu - `s`pawning - `v`ehicle - test123 - undamaged
11. Let's see the result
![2](https://github.com/user-attachments/assets/d37c5621-f4cd-40d8-9e12-786e19489df6)

### Expected behavior

n/a

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043 (21H1)
- Game Version: cdda-experimental-2024-09-27-0529 918a370 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

**Comment by SurFlurer** [comment]
Would you like to test whether this bug started to happen before 2024-09-16 or not?

**Comment by IdleSol** [comment]
```
 DEBUG    : init_vehicles: 'test123' part 'hdframe'(31) can't be installed to -3,-1

 FUNCTION : void vehicles::finalize_prototypes()
 FILE     : src/veh_type.cpp
 LINE     : 1591
 VERSION  : cdda-experimental-2024-08-01-0043 1fd261f
```

**Comment by IdleSol** [comment]
Why is there no error with the “original” truck? Because the order of parts in the trucks.json file is different from the order in the export file.

The export file sorts (maybe) by X coordinates, then Y coordinates. From negative numbers, to positive numbers. That is, in ascending order.

The file with the original truck starts from coordinates 0.0. But in this variant, we just got lucky.

If we talk about theoretical ways of fixing, then:
1. Adding fake parts on empty tiles if they are bordered by vehicle parts. These fake parts can provide support and do not participate in the vehicle integrity check. Once spawning is complete, are replaced with open space
2. Modifying the vehicle integrity check. Read all tiles first, then check.

**Comment by Joemanuel123** [comment]
![Captura de pantalla 2024-12-01 110019](https://github.com/user-attachments/assets/f7eb843c-8678-4ba0-a478-d67b399e233c)
same issue here, I'm trying to make a fighter in aftershock but this bug is slowing me down

I don't even know where to start fixing this 😅

**Comment by kevingranade** [comment]
There's a reason import/export is debug, it's complicated and weird and no one has put a ton of work into it. The expectation is it just gives you a starting point and you fix the ordering issues yourself.

Debug features are not supported and issues about debug issues are invalid.

**Comment by Joemanuel123** [comment]
Oh okay, thank you for clarifying anyway. have a wonderful day man.

**Comment by IdleSol** [comment]
@Joemanuel123 As a temporary solution. You can change the order of lines in the export file. This will help with vehicle testing, although it won't fix the problem.
